### PR TITLE
[Ldap] Add write support for the Ldap component

### PR DESCRIPTION
**Pull request by csarrazi**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | yes |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #17679 |
| License | MIT |
| Doc PR | no |

This PR implements write support for the Ldap component.

### CODE REVIEW

**Comment by fabpot** [comment]
@csarrazi Can you submit a different PR for the fixes of #16665? As those changes are non-controversial, we can merge them much faster and it won't "pollute" the new feature added here.

**Comment by csarrazi** [comment]
No problem. I'll patch-commit things, and submit 2 PRs.

**Comment by csarrazi** [comment]
Done. Ping @fabpot

**Comment by fabpot** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @csarrazi.

**Comment by L0rD59** [comment]
:+1:

File: src/Symfony/Component/Ldap/Entry.php [hunk 1]
```diff
@@ -59,4 +59,25 @@ public function getAttributes()
     {
         return $this->attributes;
     }
+
+    /**
+     * Sets a value for the given attribute.
+     *
+     * @param $name
```
Comment (xabbuh) [inline src/Symfony/Component/Ldap/Entry.php]: missing type

File: src/Symfony/Component/Ldap/Entry.php [hunk 2]
```diff
@@ -59,4 +59,25 @@ public function getAttributes()
...
+     */
+    public function setAttribute($name, array $value)
+    {
+        $this->attributes[$name] = $value;
+    }
+
+    /**
+     * Removes a given attribute.
+     *
+     * @param $name
```
Comment (xabbuh) [inline src/Symfony/Component/Ldap/Entry.php]: same here

**Pull request merged**